crates/ty_ide/src/inlay_hints.rs

```diff
@@ -1,62 +1,139 @@
+use std::fmt;
+
 use crate::Db;
 use ruff_db::files::File;
 use ruff_db::parsed::parsed_module;
 use ruff_python_ast::visitor::source_order::{self, SourceOrderVisitor, TraversalSignal};
 use ruff_python_ast::{AnyNodeRef, Expr, Stmt};
 use ruff_text_size::{Ranged, TextRange, TextSize};
-use std::fmt;
-use std::fmt::Formatter;
 use ty_python_semantic::types::{Type, inlay_hint_function_argument_details};
 use ty_python_semantic::{HasType, SemanticModel};
 
-#[derive(Debug, Clone, Eq, PartialEq)]
-pub struct InlayHint<'db> {
+#[derive(Debug, Clone)]
+pub struct InlayHint {
     pub position: TextSize,
-    pub content: InlayHintContent<'db>,
+    pub kind: InlayHintKind,
+    pub label: InlayHintLabel,
 }
 
-impl<'db> InlayHint<'db> {
-    pub const fn display(&self, db: &'db dyn Db) -> DisplayInlayHint<'_, 'db> {
-        self.content.display(db)
+impl InlayHint {
+    pub fn type_hint(position: TextSize, ty: Type, db: &dyn Db) -> Self {
+        let mut label = InlayHintLabel::simple(ty.display(db).to_string(), None);
+
+        label.prepend_str(": ");
+
+        Self {
+            position,
+            kind: InlayHintKind::Type,
+            label,
+        }
+    }
+
+    pub fn call_argument_name(position: TextSize, name: &str) -> Self {
+        let mut label = InlayHintLabel::simple(name.to_string(), None);
+
+        label.append_str("=");
+
+        Self {
+            position,
+            kind: InlayHintKind::CallArgumentName,
+            label,
+        }
+    }
+
+    pub fn display(&self) -> InlayHintDisplay<'_> {
+        InlayHintDisplay { inlay_hint: self }
     }
 }
 
-#[derive(Debug, Clone, Eq, PartialEq)]
-pub enum InlayHintContent<'db> {
-    Type(Type<'db>),
-    CallArgumentName(String),
+#[derive(Debug, Clone)]
+pub enum InlayHintKind {
+    Type,
+    CallArgumentName,
+}
+
+#[derive(Debug, Clone)]
+pub struct InlayHintLabel {
+    pub parts: Vec<InlayHintLabelPart>,
+}
```

Comment:
Are you thinking of expanding `InlayHintLabel` with additional fields in the future?

Regardless, I think the `parts` field should be private and we should expose this information via a method that returns `&[InlayHintLabelPart]` instead.